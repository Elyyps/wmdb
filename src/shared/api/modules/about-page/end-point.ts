import { generateAboutPageDummyData } from "./generate-dummy-data";

export const generateAboutPageData = generateAboutPageDummyData;
