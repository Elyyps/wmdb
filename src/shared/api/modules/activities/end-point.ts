import { generateActivitiesDummyData } from "./dummy-data";

export const generateActivitiesData = generateActivitiesDummyData;
