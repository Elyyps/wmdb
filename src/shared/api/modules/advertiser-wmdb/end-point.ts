import { generateAdvertiserDummyData } from "./dummy-data";
export const generateAdvertiserData = generateAdvertiserDummyData;
