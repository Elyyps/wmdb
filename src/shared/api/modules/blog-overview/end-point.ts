import { generateBlogOverviewDummyData, generateCards } from "./dummy-data";
import { ICompanyOutingCard } from "@app/api/core/company-outing-card";

export const generateBlogOverviewData = generateBlogOverviewDummyData;
const amountOfCards = 50;
export const getCardsWithPagination = (skip: number, take: number): ICompanyOutingCard[] => {
  let cards: ICompanyOutingCard[] = generateCards(amountOfCards);
  cards = cards.slice(skip, take + skip);

  return cards;
};
