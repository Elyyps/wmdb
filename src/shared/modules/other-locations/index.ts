export * from "./other-locations.component";
