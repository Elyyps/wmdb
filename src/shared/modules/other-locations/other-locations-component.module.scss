.other-locations {
  &:not(:first-child) {
    margin-top: 42px;
  }

  &__list {
    font-size: 12px;
    @media (min-width: $breakpoint-small) {
      column-count: 3;
      column-gap: 20px;
    }
    @media (min-width: $breakpoint-medium) {
      column-count: 5;
      font-size: 14px;
    }
    li {
      display: table;
      margin-bottom: 12px;
      text-decoration: underline;
      @media (min-width: $breakpoint-small) {
        margin-bottom: 9px;
      }
      @media (max-width: $breakpoint-xsmall-max) {
        &:nth-child(n + 5) {
          display: none;
        }
      }
    }
    a {
      color: $gray;
    }
  }
}
