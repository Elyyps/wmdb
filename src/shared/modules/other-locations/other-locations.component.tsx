import * as React from "react";
import styles from "./other-locations-component.module.scss";
import { IOtherLocationsModule } from "@app/api/modules/other-locations/other-locations";
import { ILocation } from "@app/api/core/location";
import { LinkComponent } from "@app/core/link";
import { ILink } from "@app/api/core/link";
import { ModuleSectionComponent } from "@app/core/module-section";

export interface IOtherLocationsComponentProps {
  otherLocationsModule: IOtherLocationsModule;
}

const OtherLocationsComponent = ({ otherLocationsModule }: IOtherLocationsComponentProps) => (
  <ModuleSectionComponent
    title="Andere locaties"
    backgroundColor={"#e8f2fc"}
    paddingTop={"0px"}
    paddingTopMobile="0px"
    paddingBottom={"72px"}
  >
    {otherLocationsModule.locations.map((locations: ILocation, key: number) => (
      <div className={styles["other-locations"]} key={key}>
        <h5>{locations.title}</h5>
        {locations.links && (
          <ul className={styles["other-locations__list"]}>
            {locations.links.map((link: ILink, i: number) => (
              <li key={i}>
                <LinkComponent to={link.url}>{link.title}</LinkComponent>
              </li>
            ))}
          </ul>
        )}
      </div>
    ))}
  </ModuleSectionComponent>
);

export { OtherLocationsComponent };
