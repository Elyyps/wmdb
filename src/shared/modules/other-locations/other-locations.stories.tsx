import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { OtherLocationsComponent } from "./other-locations.component";
import { generateOtherLocationsData } from "@app/api/modules/other-locations/end-point";

storiesOf("OtherLocations", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <OtherLocationsComponent otherLocationsModule={generateOtherLocationsData()} />);
