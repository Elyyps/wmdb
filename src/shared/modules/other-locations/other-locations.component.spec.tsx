import React from "react";
import { shallow } from "enzyme";
import { OtherLocationsComponent } from "./other-locations.component";
import { generateOtherLocationsData } from "@app/api/modules/other-locations/end-point";

describe("[OtherLocations]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<OtherLocationsComponent otherLocations={generateOtherLocationsData()} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
