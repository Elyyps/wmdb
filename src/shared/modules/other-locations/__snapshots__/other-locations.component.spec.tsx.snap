// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[OtherLocations] should render component without crashing 1`] = `
<ModuleSectionComponent
  backgroundColor="#e8f2fc"
  paddingBottom="72px"
  paddingTop="0px"
  paddingTopMobile="0px"
  title="Andere locaties"
>
  <div
    className="other-locations"
    key="0"
  >
    <h5>
      Locaties per plaats
    </h5>
    <ul
      className="other-locations__list"
    >
      <li
        key="0"
      >
        <LinkComponent
          to=""
        >
          Locaties Egmond aan Zee
        </LinkComponent>
      </li>
      <li
        key="1"
      >
        <LinkComponent
          to=""
        >
          Locaties Vijfhuizen
        </LinkComponent>
      </li>
      <li
        key="2"
      >
        <LinkComponent
          to=""
        >
          Locaties Den Dolder
        </LinkComponent>
      </li>
      <li
        key="3"
      >
        <LinkComponent
          to=""
        >
          Locaties Kollumerpomp
        </LinkComponent>
      </li>
      <li
        key="4"
      >
        <LinkComponent
          to=""
        >
          Locaties Egmond aan Zee
        </LinkComponent>
      </li>
      <li
        key="5"
      >
        <LinkComponent
          to=""
        >
          Locaties Vijfhuizen
        </LinkComponent>
      </li>
      <li
        key="6"
      >
        <LinkComponent
          to=""
        >
          Locaties Den Dolder
        </LinkComponent>
      </li>
      <li
        key="7"
      >
        <LinkComponent
          to=""
        >
          Locaties Kollumerpomp
        </LinkComponent>
      </li>
      <li
        key="8"
      >
        <LinkComponent
          to=""
        >
          Locaties Egmond aan Zee
        </LinkComponent>
      </li>
      <li
        key="9"
      >
        <LinkComponent
          to=""
        >
          Locaties Vijfhuizen
        </LinkComponent>
      </li>
      <li
        key="10"
      >
        <LinkComponent
          to=""
        >
          Locaties Den Dolder
        </LinkComponent>
      </li>
      <li
        key="11"
      >
        <LinkComponent
          to=""
        >
          Locaties Kollumerpomp
        </LinkComponent>
      </li>
      <li
        key="12"
      >
        <LinkComponent
          to=""
        >
          Locaties Egmond aan Zee
        </LinkComponent>
      </li>
      <li
        key="13"
      >
        <LinkComponent
          to=""
        >
          Locaties Vijfhuizen
        </LinkComponent>
      </li>
      <li
        key="14"
      >
        <LinkComponent
          to=""
        >
          Locaties Den Dolder
        </LinkComponent>
      </li>
      <li
        key="15"
      >
        <LinkComponent
          to=""
        >
          Locaties Kollumerpomp
        </LinkComponent>
      </li>
      <li
        key="16"
      >
        <LinkComponent
          to=""
        >
          Locaties Egmond aan Zee
        </LinkComponent>
      </li>
      <li
        key="17"
      >
        <LinkComponent
          to=""
        >
          Locaties Vijfhuizen
        </LinkComponent>
      </li>
      <li
        key="18"
      >
        <LinkComponent
          to=""
        >
          Locaties Den Dolder
        </LinkComponent>
      </li>
      <li
        key="19"
      >
        <LinkComponent
          to=""
        >
          Locaties Kollumerpomp
        </LinkComponent>
      </li>
    </ul>
  </div>
  <div
    className="other-locations"
    key="1"
  >
    <h5>
      Locaties per provincie
    </h5>
    <ul
      className="other-locations__list"
    >
      <li
        key="0"
      >
        <LinkComponent
          to=""
        >
          Locaties Egmond aan Zee
        </LinkComponent>
      </li>
      <li
        key="1"
      >
        <LinkComponent
          to=""
        >
          Locaties Vijfhuizen
        </LinkComponent>
      </li>
      <li
        key="2"
      >
        <LinkComponent
          to=""
        >
          Locaties Den Dolder
        </LinkComponent>
      </li>
      <li
        key="3"
      >
        <LinkComponent
          to=""
        >
          Locaties Kollumerpomp
        </LinkComponent>
      </li>
      <li
        key="4"
      >
        <LinkComponent
          to=""
        >
          Locaties Egmond aan Zee
        </LinkComponent>
      </li>
      <li
        key="5"
      >
        <LinkComponent
          to=""
        >
          Locaties Vijfhuizen
        </LinkComponent>
      </li>
      <li
        key="6"
      >
        <LinkComponent
          to=""
        >
          Locaties Den Dolder
        </LinkComponent>
      </li>
      <li
        key="7"
      >
        <LinkComponent
          to=""
        >
          Locaties Kollumerpomp
        </LinkComponent>
      </li>
      <li
        key="8"
      >
        <LinkComponent
          to=""
        >
          Locaties Egmond aan Zee
        </LinkComponent>
      </li>
      <li
        key="9"
      >
        <LinkComponent
          to=""
        >
          Locaties Vijfhuizen
        </LinkComponent>
      </li>
      <li
        key="10"
      >
        <LinkComponent
          to=""
        >
          Locaties Den Dolder
        </LinkComponent>
      </li>
      <li
        key="11"
      >
        <LinkComponent
          to=""
        >
          Locaties Kollumerpomp
        </LinkComponent>
      </li>
      <li
        key="12"
      >
        <LinkComponent
          to=""
        >
          Locaties Egmond aan Zee
        </LinkComponent>
      </li>
      <li
        key="13"
      >
        <LinkComponent
          to=""
        >
          Locaties Vijfhuizen
        </LinkComponent>
      </li>
      <li
        key="14"
      >
        <LinkComponent
          to=""
        >
          Locaties Den Dolder
        </LinkComponent>
      </li>
      <li
        key="15"
      >
        <LinkComponent
          to=""
        >
          Locaties Kollumerpomp
        </LinkComponent>
      </li>
      <li
        key="16"
      >
        <LinkComponent
          to=""
        >
          Locaties Egmond aan Zee
        </LinkComponent>
      </li>
      <li
        key="17"
      >
        <LinkComponent
          to=""
        >
          Locaties Vijfhuizen
        </LinkComponent>
      </li>
      <li
        key="18"
      >
        <LinkComponent
          to=""
        >
          Locaties Den Dolder
        </LinkComponent>
      </li>
      <li
        key="19"
      >
        <LinkComponent
          to=""
        >
          Locaties Kollumerpomp
        </LinkComponent>
      </li>
    </ul>
  </div>
  <div
    className="other-locations"
    key="2"
  >
    <h5>
      Locaties per speltype
    </h5>
    <ul
      className="other-locations__list"
    >
      <li
        key="0"
      >
        <LinkComponent
          to=""
        >
          Locaties Egmond aan Zee
        </LinkComponent>
      </li>
      <li
        key="1"
      >
        <LinkComponent
          to=""
        >
          Locaties Vijfhuizen
        </LinkComponent>
      </li>
      <li
        key="2"
      >
        <LinkComponent
          to=""
        >
          Locaties Den Dolder
        </LinkComponent>
      </li>
      <li
        key="3"
      >
        <LinkComponent
          to=""
        >
          Locaties Kollumerpomp
        </LinkComponent>
      </li>
      <li
        key="4"
      >
        <LinkComponent
          to=""
        >
          Locaties Egmond aan Zee
        </LinkComponent>
      </li>
      <li
        key="5"
      >
        <LinkComponent
          to=""
        >
          Locaties Vijfhuizen
        </LinkComponent>
      </li>
      <li
        key="6"
      >
        <LinkComponent
          to=""
        >
          Locaties Den Dolder
        </LinkComponent>
      </li>
      <li
        key="7"
      >
        <LinkComponent
          to=""
        >
          Locaties Kollumerpomp
        </LinkComponent>
      </li>
      <li
        key="8"
      >
        <LinkComponent
          to=""
        >
          Locaties Egmond aan Zee
        </LinkComponent>
      </li>
      <li
        key="9"
      >
        <LinkComponent
          to=""
        >
          Locaties Vijfhuizen
        </LinkComponent>
      </li>
      <li
        key="10"
      >
        <LinkComponent
          to=""
        >
          Locaties Den Dolder
        </LinkComponent>
      </li>
      <li
        key="11"
      >
        <LinkComponent
          to=""
        >
          Locaties Kollumerpomp
        </LinkComponent>
      </li>
      <li
        key="12"
      >
        <LinkComponent
          to=""
        >
          Locaties Egmond aan Zee
        </LinkComponent>
      </li>
      <li
        key="13"
      >
        <LinkComponent
          to=""
        >
          Locaties Vijfhuizen
        </LinkComponent>
      </li>
      <li
        key="14"
      >
        <LinkComponent
          to=""
        >
          Locaties Den Dolder
        </LinkComponent>
      </li>
      <li
        key="15"
      >
        <LinkComponent
          to=""
        >
          Locaties Kollumerpomp
        </LinkComponent>
      </li>
      <li
        key="16"
      >
        <LinkComponent
          to=""
        >
          Locaties Egmond aan Zee
        </LinkComponent>
      </li>
      <li
        key="17"
      >
        <LinkComponent
          to=""
        >
          Locaties Vijfhuizen
        </LinkComponent>
      </li>
      <li
        key="18"
      >
        <LinkComponent
          to=""
        >
          Locaties Den Dolder
        </LinkComponent>
      </li>
      <li
        key="19"
      >
        <LinkComponent
          to=""
        >
          Locaties Kollumerpomp
        </LinkComponent>
      </li>
    </ul>
  </div>
</ModuleSectionComponent>
`;
